packages/core-snapshots/src/codecs/codec.ts

```diff
@@ -0,0 +1,125 @@
+import { Models } from "@arkecosystem/core-database";
+import { Container } from "@arkecosystem/core-kernel";
+import { Blocks, Interfaces, Transactions, Utils } from "@arkecosystem/crypto";
+import { decode, encode } from "msgpack-lite";
+import { camelizeKeys } from "xcase";
+
+import { Codec as ICodec } from "../contracts";
+import { Codec as CodecException } from "../exceptions";
+
+@Container.injectable()
+export class Codec implements ICodec {
+    private static removePrefix(item: Record<string, any>, prefix: string): Record<string, any> {
+        const itemToReturn = {};
+
+        for (const key of Object.keys(item)) {
+            itemToReturn[key.replace(prefix, "")] = item[key];
+        }
+
+        return itemToReturn;
+    }
+
+    public blocksEncode(block: any) {
+        try {
+            const blockCamelized = camelizeKeys(Codec.removePrefix(block, "Block_"));
+
+            return Blocks.Serializer.serialize(blockCamelized, true);
+        } catch (e) {
+            throw new CodecException.BlockEncodeException(block.Block_id);
+        }
+    }
+
+    public blocksDecode(buffer: Buffer) {
+        let blockId = undefined;
+        try {
+            const block: any = Blocks.Deserializer.deserialize(buffer.toString("hex"), false).data;
+
+            blockId = block.id;
+
+            return block;
+        } catch (e) {
+            throw new CodecException.BlockDecodeException((blockId as unknown) as string);
+        }
+    }
+
+    public transactionsEncode(transaction) {
+        try {
+            return encode([
+                transaction.Transaction_id,
+                transaction.Transaction_block_id,
+                transaction.Transaction_sequence,
+                transaction.Transaction_timestamp,
+                transaction.Transaction_serialized,
+            ]);
+        } catch (e) {
+            throw new CodecException.TransactionEncodeException(transaction.Transaction_id);
+        }
+    }
+
+    public transactionsDecode(buffer: Buffer) {
+        let transactionId = undefined;
+        try {
+            const [id, blockId, sequence, timestamp, serialized] = decode(buffer);
+            transactionId = id;
+
+            const transaction: Interfaces.ITransaction = Transactions.TransactionFactory.fromBytesUnsafe(
+                serialized,
+                id,
+            );
+
+            const transactionEntity: Models.Transaction = {
```

Comment:
Return this, no point in assigning it to a variable and then immediately returning.